### ISSUE TITLE
Fractional-width image overlay/scaling

### ISSUE THREAD

**Issue by jpolitz** [brownplt/code.pyret.org#320]
Consider this program:

```
include image


fun trap(base1, base2, height, clr):
  above(
    (right-triangle(height, (base2 - base1), "solid", clr)),
    rectangle(height, base1 , "solid", clr))
end

trap(50, 60, 4, "red")
scale(10, trap(50, 60, 4, "red"))

trap(20, 30, 0.5, "blue")
scale(10, trap(20, 30, 0.5, "blue"))
```

The output looks like this:

![image](https://user-images.githubusercontent.com/119891/70158174-a859f180-166b-11ea-9b4d-985c19b016bb.png)

Note that the trapezoid created with the fractional width is mis-aligned. I think it's mis-aligned in the smaller image, too. This doesn't seem to happen for fractional widths greater than 1.

**Comment by blerner** [comment]
This holds true for the horizontal version of this code as well.  
```
fun htrap(base1, base2, height, clr):
  beside(
    rectangle(base1, height, "solid", clr),
    (right-triangle((base2 - base1), height, "solid", clr)))
end

scale(30, htrap(20, 30, 0.5, "blue"))
```
![image](https://user-images.githubusercontent.com/918464/70191830-087e8100-16c8-11ea-8c0c-f80b7e6da24b.png)



I suspect (but haven't confirmed yet) that there's a 1px rounding for rectangles happening that we don't quite do in the same way for triangles.

**Comment by blerner** [comment]
This has been fixed by #267 (see https://github.com/brownplt/code.pyret.org/commit/b80b60c70509d0e3223f458b318da07c2e390a4f)
